File: Python/src/square_digits.py
def square_digits(num: int) -> int:
    num_string = ''
    for c in str(num):
        num_string += str(int(c)**2)
    return int(num_string)


def square_digits_v2(num: int) -> int:
    return int(''.join
               (map(lambda c: str(int(c)**2), str(num))))

File: Typescript/src/squareDigits.ts
export function squareDigits(num: number): number {
    let squaredNumString = '';
    for (const c of num.toString()) {
        squaredNumString += Number(c) ** 2;
    }
    return Number(squaredNumString);
}

export function squareDigitsV2(num: number): number {
    return Number(num.toString().split('').map(c => Number(c) ** 2).join(''));
    // + converts a string to an integer
    return +num.toString().split('').map(c => +c * +c).join('');
}